apps/app_rpt/rpt_config.c

```diff
@@ -1354,3 +1359,36 @@ void rpt_update_boolean(struct rpt *myrpt, char *varname, int newval)
 		rpt_event_process(myrpt);
 	}
 }
+
+bool rpt_is_valid_dns_name(const char *dns_name) {
+	if (!dns_name || strlen(dns_name) > MAX_DOMAIN_LENGTH) {
+		return false;
+	}
+
+	int label_length = 0;
+	bool label_start = true;
+
+	for (const char *ptr = dns_name; *ptr; ++ptr) {
+		if (*ptr == '.') {
+			if (label_start || label_length == 0) {
```

Comment:
Is it possible for one or the other here?  I think `if (label_start) { ` is sufficient to cover all of the cases (and make it a bit easier to understand).
Essentially, if we have a `.` and `label_start == TRUE`, there was nothing between the last `.` and this time around.